interview-questions.md

```diff
@@ -7,53 +7,53 @@ Then, research the question to expand on your answer. Even if you feel you have
 
 1. What is a block in Ruby?
 
-  Your answer:
+  Your answer: Ruby blocks are functions that could be passed into methods. Take arguments which would be between two pipes.
 
-  Researched answer:
+  Researched answer: Ruby blocks have anonymous functions that can be passed into methods. To me I like to think of it as a function that I use in javascript. Just a little upside down. It could have arguments as well and they live between pipes which is something so different from JS.
 
 
 
 2. What is a gem?
 
-  Your answer:
+  Your answer: Gem is a package manager for Ruby programming language.
 
-  Researched answer:
+  Researched answer: Gem is a library of code. Gem is a package manager for Ruby programming language. It's really weird that this sums up all of what I could find online.
 
 
 
 3. What is Ruby on Rails?
 
-  Your answer:
+  Your answer: It's a software that programmers can use to build web applications.
 
-  Researched answer:
+  Researched answer: It's a software that you can use to create web applications front end and back end. Something that I didn't know was that a framework makes its easier because it gives you the structure of the code already built in. It has the MVC, which stands for the model, the view, and the controller.
 
 
 
 4. What is a relational database? Are there other kinds of databases?
 
-  Your answer:
+  Your answer: Relational database is being able to represent data in a table. I do forget if there are others.
 
-  Researched answer:
+  Researched answer: relational database are really good in the sense that as a programmer you can call on data really easily. You can see how each data works with one another. It was fun for me especially to see how everything works. Use SQL to communicate with the relational database.
 
 
 
 5. What are primary keys? Why are they important?
 
-  Your answer:
+  Your answer: I know that the primary keys are important in relation to databases. There are so many records that can be placed inside the database that without the primary key. It would be hard to sort things out.
 
-  Researched answer:
+  Researched answer: The reason they are so important because you use the key to look or compare the keys that hold that data/record. One advice that is very important to follow is keep it short, keep it simple, and never change. The table is sorted based on primary key.
```

Comment:
Primary keys are unique identifiers for your columns in a database.  Because they are unique, data can have similar or the same values inside the table but can be a unique entry because of the primary keys.